src/components/color_picker/color_picker_entry.ts

```diff
@@ -0,0 +1,49 @@
+import { Component } from "@odoo/owl";
+import { SpreadsheetChildEnv } from "../../types";
+import { css } from "../helpers";
+import { ColorPicker } from "./color_picker";
+
+interface Props {
+  currentColor: string | undefined;
+  onClick: () => void;
+  showColorPicker: boolean;
+  onColorPicked: (color: string) => void;
+  icon: string;
+  dropdownDirection?: "left" | "right" | "center";
+  tooltipText?: string;
+}
+
+css/* scss */ `
+  .o-fill-color-entry span {
+    cursor: pointer;
+  }
+
+  .o-cf-cell-is-rule .o-fill-color-entry span {
+    cursor: default;
+  }
+`;
+
+export class ColorPickerEntry extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-ColorPickerEntry";
+  static components = { ColorPicker };
+
+  get iconStyle() {
+    return this.props.currentColor
+      ? `border-color: ${this.props.currentColor}`
+      : "border-bottom-style: hidden";
+  }
+
+  get dropdownDirection() {
+    return this.props.dropdownDirection || "right";
+  }
+}
+
+ColorPickerEntry.props = {
+  currentColor: { type: String, optional: true },
```

Comment:
This should not be `optional: true` from a semantic POV, but I don't know how to make it receive `undefined` value here.

Reply:
I'm pretty sure `optional: true` is the correct way to write it, `| udnefined` and `optional` are the same fro a JS POV, even if I do agree that sometimes `| undefined` is better to understand the code